tests/puppeteer/pages/BO/products.js

```diff
@@ -123,4 +134,55 @@ module.exports = class Product extends BOBasePage {
       this.page.click(this.addProductButton),
     ]);
   }
+
+  /**
+   * Delete product with dropdown Menu
+   * @param productData
+   * @return {Promise<textContent>}
+   */
+  async deleteProduct(productData) {
+    // Filter By reference first
+    await this.filterProducts('reference', productData.reference);
+    // Then delete first product and only product shown
+    await Promise.all([
+      this.page.waitForSelector(`${this.dropdownToggleButton}[aria-expanded='true']`.replace('%ROW', '1')),
```

Comment:
I don't understand the `.replace` part ?

Reply:
`this.dropdownToggleButton` exists in all rows,, `replace` will show us which row to use.